[Tutorials][Docenhancement][1635893] (#2927)
MicrosoftDocs/msteams-docs: msteams-platform/tutorials/get-started-yeoman.md

### Install a code editor

You need a code editor. Most of this documentation and images refer to using [Visual Studio Code](https://code.visualstudio.com). However, feel free to use whatever text editor you prefer.

### Install Yeoman and Gulp CLI

> To be able to scaffold projects using the Teams generator you need to install the Yeoman tool and the Gulp CLI task manager.

Reviewer: ```suggestion
To scaffold projects using the generator, you must install the Yeoman tool and Gulp CLI task manager.
```

npm install generator-teams --global
```

Install the preview version of the generator with the following command:

```bash
npm install generator-teams@preview --global
```

## Generate your project
This section walks you through the steps for generating your project.

> **To generate your project**

Reviewer: @KirtiPereira Is this a preferred convention now? I believe the docs validation tool flags this since it's not a semantic heading (H1, H2, etc.). It also repeats the heading above (Generate your project).

Author: Hi @heath-hamilton - yes, we are going with this convention, wherever we are calling out the steps. Will check about the validation tool flags, as we not received any warning or faced an issue.

   > [!NOTE]
   > * For this tutorial choose not to include a test framework.
   > * The generator has a lot of built-in advanced features that you can opt-in or opt-out of.

1. In order to make signing-in easy for you, you will also be asked if you want to use Azure Application Insights for signing-in. If you choose *Yes*, you will need to provide an Azure Application Insights key. 

   > [!NOTE]
   > For this tutorial opt-out of using Application Insights.

The next set of questions will be based on the previously selected items. For a tab you only need to provide a name and optionally choose if you want to be able to use this app as a SharePoint Online web part. After you provide the name the generator will generate the project and install all dependencies. This will take a minute or two.

## Add some code to your tab

After the generator is done you can open up the solution in your favorite code editor. Take a minute or two and familiarize yourself with how the code is organized. For more information, see [Project Structure](https://github.com/OfficeDev/generator-teams/wiki/Project-Structure) documentation.

> Your tab will be located in the `./src/app/scripts/myFirstAppTab/MyFirstAppTab.tsx` file. This is the TypeScript React based class for your Tab. Locate the `render()` method and add a line of code inside the `<PanelBody>` control so it looks like this:

Reviewer: ```suggestion
Your tab is in the `./src/app/scripts/myFirstAppTab/MyFirstAppTab.tsx` file. This is the TypeScript React-based class for your tab. Locate the `render()` method and add a line of code inside the `<PanelBody>` control so it looks like this:
```

![running tab in teams](~/assets/yeoman-images/teams-first-app-6.png)

> ## Upgrade Microsoft Teams

Reviewer: @KirtiPereira Docs validation tool generates a warning if you do not have a space between a header and body text (or other element).

You can also upgrade your current Microsoft Teams version to the latest version using the Microsoft Teams Yeoman generator.

> **To upgrade Microsoft Teams**

Reviewer: @KirtiPereira I do not think this is needed when the heading already conveys this.

Author: @heath-hamilton - Will keep this for now. Per MSTP convention, it is good to have a procedural heading. 
Thanks.
